src/game/Spells/SpellAuras.cpp

```diff
@@ -6234,7 +6234,14 @@ void Aura::PeriodicDummyTick()
                 }
                 case 8067:                                  // Party Time!
                 {
-                    target->HandleEmoteCommand(PickRandomValue(EMOTE_ONESHOT_APPLAUD, EMOTE_ONESHOT_CHEER, EMOTE_ONESHOT_CHICKEN, EMOTE_ONESHOT_LAUGH, EMOTE_ONESHOT_DANCE));
+                    if (!target->HasAura(8067))
+                    {
+                        if (target->IsMoving())
+                            target->HandleEmoteCommand(PickRandomValue(EMOTE_ONESHOT_APPLAUD, EMOTE_ONESHOT_CHEER, EMOTE_ONESHOT_CHICKEN, EMOTE_ONESHOT_LAUGH));
+                        else
+                            target->HandleEmoteCommand(PickRandomValue(EMOTE_ONESHOT_APPLAUD, EMOTE_ONESHOT_CHEER, EMOTE_ONESHOT_CHICKEN, EMOTE_ONESHOT_LAUGH, EMOTE_ONESHOT_DANCESPECIAL));
+                        // TODO: TrintiyCore: _player->m_Events.AddEventAtOffset(this, RAND(5s, 10s, 15s));
+                    }
```

Comment:
What are you trying to do here?

Reply:
Removed this as well. I think trinity added a random time before the "party emote event" gets triggered on player.

Reply:
Due to https://vanilla-wow.fandom.com/wiki/Deviate_Fish, the party emote lasts 2 minutes and gets triggered every 10–15 seconds. That's probaly what they what trinity did with `AddEventAtOffset`.